### ISSUE TITLE
Ruby 3.3 doesn't work

### ISSUE THREAD

**Issue by Stashchenko** [dependabot/dependabot-core#8739]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

bundler

### Package manager version

2.4.20 or 2.5.4

### Language version

3.3.0

### Manifest location and content before the Dependabot update

```ruby
# frozen_string_literal: true

source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '3.3.0'

# Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
gem 'rails', '~> 7.1'

# Use postgresql as the database for Active Record
gem 'pg', '~> 1.5'

# Use the Puma web server [https://github.com/puma/puma]
gem 'puma', '~> 6.3'
```

### dependabot.yml content

```yaml
updates:
  - package-ecosystem: "bundler"
    directory: "/"
    registries: "*"
    insecure-external-code-execution: allow
    schedule:
      interval: "weekly"
      day: "sunday"
    groups:
      dependencies:
        patterns:
          - "*"       # A wildcard that matches all dependencies in the package
```

### What you expected to see, versus what you actually saw

when we updated to the new ruby

### Images of the diff or a link to the PR, issue, or logs

```shell
pdater | 2024/01/07 19:05:44 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:64:in `run'
updater | 2024/01/07 19:05:44 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/update_files_command.rb:43:in `perform_job'
updater | 2024/01/07 19:05:44 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/base_command.rb:53:in `run'
updater | 2024/01/07 19:05:44 ERROR <job_770657921> bin/update_files.rb:24:in `<main>'
updater | 2024/01/07 19:05:45 INFO <job_770657921> Checking if store_model 2.1.2 needs updating
updater | 2024/01/07 19:05:45 INFO <job_770657921> Sending event 42bcf9bb8aa241599bd78b0d6532acfe to Sentry
updater | 2024/01/07 19:05:45 ERROR <job_770657921> Error processing store_model (Dependabot::Bundler::FileUpdater::RubyRequirementSetter::RubyVersionNotFound)
updater | 2024/01/07 19:05:45 ERROR <job_770657921> Dependabot::Bundler::FileUpdater::RubyRequirementSetter::RubyVersionNotFound
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/file_updater/ruby_requirement_setter.rb:65:in `ruby_version'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/file_updater/ruby_requirement_setter.rb:36:in `rewrite'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker/file_preparer.rb:254:in `block in lock_ruby_version'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker/file_preparer.rb:252:in `each'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker/file_preparer.rb:252:in `lock_ruby_version'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker/file_preparer.rb:170:in `gemfile_content_for_update_check'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker/file_preparer.rb:62:in `prepared_dependency_files'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker.rb:382:in `prepared_dependency_files'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker.rb:356:in `latest_version_finder'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker.rb:197:in `latest_version_details'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/update_checker.rb:22:in `latest_version'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/group_update_creation.rb:212:in `all_versions_ignored?'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/group_update_creation.rb:134:in `compile_updates_for'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/group_update_creation.rb:48:in `block in compile_all_dependency_changes_for'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/group_update_creation.rb:29:in `each'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/group_update_creation.rb:29:in `compile_all_dependency_changes_for'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/create_group_update_pull_request.rb:70:in `dependency_change'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/create_group_update_pull_request.rb:44:in `perform'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:117:in `run_update_for'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:99:in `block in run_grouped_dependency_updates'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:98:in `each'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:98:in `run_grouped_dependency_updates'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:45:in `perform'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:64:in `run'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/update_files_command.rb:43:in `perform_job'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> /home/dependabot/dependabot-updater/lib/dependabot/base_command.rb:53:in `run'
updater | 2024/01/07 19:05:45 ERROR <job_770657921> bin/update_files.rb:24:in `<main>'
updater | 2024/01/07 19:05:46 INFO <job_770657921> Checking if rails 7.1.2 needs updating
updater | 2024/01/07 19:05:46 INFO <job_770657921> Sending event dc10840722f54476874ac5f827d0ae99 to Sentry
updater | 2024/01/07 19:05:46 ERROR <job_770657921> Error processing rails (Dependabot::Bundler::FileUpdater::RubyRequirementSetter::RubyVersionNotFound)
updater | 2024/01/07 19:05:46 ERROR <job_770657921> Dependabot::Bundler::FileUpdater::RubyRequirementSetter::RubyVersionNotFound
updater | 2024/01/07 19:05:46 ERROR <job_770657921> /home/dependabot/bundler/lib/dependabot/bundler/file_updater/ruby_requirement_setter.rb:65:in `ruby_version'

```

**Comment by schinery** [comment]
Am here as I have the same issue in one Ruby 3.3.0 project, however I'm not 100% sure if it is because of Ruby 3.3 as Dependabot is running in another Ruby 3.3.0 project.

I haven't had time to investigate myself in the project that is failing. my initial hunch was maybe a gem had been added that was causing the issue 🤷🏻‍♂️

**Comment by Stashchenko** [comment]
> Am here as I have the same issue in one Ruby 3.3.0 project, however I'm not 100% sure if it is because of Ruby 3.3 as Dependabot is working correctly in another Ruby 3.3.0 project.
> 
> I haven't had time to investigate myself in the project that is failing. my initial hunch was maybe a gem had been added that was causing the issue 🤷🏻‍♂️

We tested the same dependbot's configuration with ruby-3.2.2 - works ok,yYet after upgrade - not working :(

**Comment by schinery** [comment]
Ah, yeah it is [this](https://github.com/dependabot/dependabot-core/blob/f913bb2d92565fca6ba91d464067f1e2f97d0e3a/bundler/lib/dependabot/bundler/file_updater/ruby_requirement_setter.rb#L15) that needs updating to include Ruby 3.3.

Generated https://github.com/dependabot/dependabot-core/pull/8754 to add it. Totally forgot I had a similar issue with Ruby 3.2.0 🤦🏻‍♂️

**Comment by deivid-rodriguez** [comment]
Closed by #8754, thanks again @schinery!
